mentorship/slack_coc.md

```diff
@@ -0,0 +1,70 @@
+
+**Messages on public channels of the Slack may be put in a public archive or shared outside the Slack**. Use DMs/private channels for private messages, which should not be shared without consent.
+
+By joining the Slack, members also agree to the following general code of conduct. The administrators will enforce this code of conduct and expect cooperation in ensuring a safe environment for all. 
+
+## Code of Conduct
+
+The [administrators](https://r-devel.slack.com/account/workspace-settings#admins) of R-dev Slack are dedicated to providing a harassment-free workspace for everyone regardless of age, gender, sexual orientation, disability, physical appearance, race, or religion (or lack thereof). 
+
+### Expected Behaviour
+
+All members agree to: 
+
+- Be considerate in language and actions, and respect the boundaries of fellow members.
+- Refrain from demeaning, discriminatory, or harassing behaviour and language.  Please refer to [Unacceptable Behaviour](#unacceptable-behaviour) for more details. 
+- Alert one of the [administrators](https://r-devel.slack.com/account/workspace-settings#admins) if you observe violations of this code of conduct, even if they seem inconsequential.
+
+### Unacceptable Behaviour
+
+Behaviour that is unacceptable includes, but is not limited to:
+
+ - Stalking, e.g. persistent unwanted DMs, logging online activity for harassment purposes.
+ - Deliberate intimidation.
+ - Sustained or willful disruption of discussion.
+ - Use of sexual or discriminatory imagery, comments, or jokes.
+ - Offensive comments related to age, gender, sexual orientation, disability, race or religion.
+ - Inappropriate simulated physical contact, e.g. textual descriptions like "hug" or "backrub", without consent.
+ - Unwelcome sexual attention, which can include inappropriate questions of a sexual nature, asking for sexual favours or repeatedly asking for dates or contact information.
+
+If you are asked to stop harassing behaviour you should stop immediately, even if your behaviour was meant to be friendly or a joke, it was clearly not taken that way and for the comfort of all members you should stop.
+
+Members who behave in a manner deemed inappropriate are subject to actions listed under [Procedure for Code of Conduct Violations](#procedure-for-code-of-conduct-violations).
+
+### Procedure for Code of Conduct Violations
+
+The administrators reserves the right to determine the appropriate response for all code of conduct violations. Potential responses include:
+
+ - a formal warning to stop harassing behaviour
+ - temporarily/permanently deactivate membership
+
+### What To Do If You Witness or Are Subject To Unacceptable Behaviour
+
+View the [Admins & Owners](https://r-devel.slack.com/account/workspace-settings#admins) in the workspace settings (only available to members). Email or message one or more administrators listed to report the behaviour.
```

Comment:
Given that it might take some time for an admin to be able to respond, I think in addition to reporting the behavior, a bystander may want to signal in the moment that the behavior is unacceptable. 

I’ll offer below some informal language for guidance that teaching assistants followed for breakout rooms at a recent R conference. I'm not sure if some version of this would be helpful to include in the Code of Conduct for everyone in the Slack, but it might be useful for admins: 

> If there’s an incident during a break-out session (chat box or vocally), the TA will respond in the moment to signal to all learners that the violation will not be tolerated (and proceed to inform [point person]).
> Responses can be brief and to the point-- the TA’s responsibility is to themselves and the learners that abide by the code of conduct. 
> Responses might include some variation of:
> - We don’t tolerate that here 
> - That’s inappropriate and unacceptable
> - Please remember you have agreed to the code of conduct

We didn’t end up having any incidents at the conference, but I at least felt better having at least some options at the ready that could be used to respond immediately.

Reply:
Another good point. We have given similar guidance to session chairs at useR!. In the Slack though, there is no such intermediate role, so we rely on individuals to respond if the admins aren't available. I added the following:

> If the admins are unable to respond in a timely manner, you may flag the issue by responding to the offending post with a brief comment, e.g.
>
> - This comment is inappropriate, please delete it.
> - This goes against the code of conduct, I have reported it.
>
> There is no obligation to respond if you don't feel comfortable. Don't get drawn into an argument or pile on an offender, a brief comment is enough to signal the issue and give the offender a chance to make it right.

I also realised that someone could delete offending posts, which could make it difficult for admins to respond/give a loophole for malicious use of the Slack. So I added advice to take screenshots of offending posts *before* contacting admins or responding.

Reply:
This is great, thank you. And great point about the screenshots, I hadn't considered that a deleted message would be deleted from the history even for admins